### PR TITLE
fix some export problems from Data.String

### PR DESCRIPTION
**Pull request by CarstenKoenig**
`Data.String` don't export `singleton : Char -> ...` and `charAt` anymore but it's in `Data.String.CodeUnits`

using these changes the package builds for me

### CODE REVIEW

**Comment by CarstenKoenig** [comment]
@michaelficarra like this (see the last commit https://github.com/purescript-contrib/purescript-argonaut-codecs/pull/42/commits/bda797910997df85edd6c85afea2bd1e500c2e25)?

test are all passing (see next comment) but I'm not sure if you want to drop the `DecodeJson Char` instance - or if you want to keep the `EncodeJson Char` one

**Comment by CarstenKoenig** [comment]
**remark** right now when I try this, I get an error from `purescript-strongcheck` at this point: https://github.com/purescript-contrib/purescript-strongcheck/blob/compiler/0.12/src/Test/StrongCheck.purs#L80

(filed an issue here: https://github.com/purescript-contrib/purescript-strongcheck/issues/51)

so I just removed the two lines locally and the project builds/tests - of course Travis still fails till this issue is fixed

**Comment by CarstenKoenig** [comment]
latest commit should make travis use purs 0.12 - of course it still fails because of the issues with `purescript-strongcheck`

File: src/Data/Argonaut/Decode/Class.purs [hunk 1]
```diff
@@ -12,7 +12,8 @@ import Data.Int (fromNumber)
 import Data.List (List(..), (:), fromFoldable)
 import Data.Map as M
 import Data.Maybe (maybe, Maybe(..))
-import Data.String (charAt)
+import Data.String (CodePoint)
+import Data.String.CodePoints (codePointAt)
```
Comment (michaelficarra) [inline src/Data/Argonaut/Decode/Class.purs]: Is this not exported by `Data.String`?

File: .travis.yml [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
-dist: trusty
 sudo: required
-node_js: stable
+dist: trusty
+node_js: 8
+env:
+  - PATH=$HOME/purescript:$PATH
 install:
+  - TAG=$(wget -q -O - https://github.com/purescript/purescript/releases/latest --server-response --max-redirect 0 2>&1 | sed -n -e 's/.*Location:.*tag\///p')
+  - wget -O $HOME/purescript.tar.gz https://github.com/purescript/purescript/releases/download/$TAG/linux64.tar.gz
```
Comment (michaelficarra) [inline .travis.yml]: I presume this is just temporary until we can bump the dependency in `package.json`?

File: src/Data/Argonaut/Encode/Class.purs [hunk 3]
```diff
@@ -48,9 +48,12 @@ instance encodeJsonJString :: EncodeJson String where
 instance encodeJsonJson :: EncodeJson Json where
   encodeJson = identity
 
-instance encodeJsonChar :: EncodeJson Char where
+instance encodeJsonCodePoint :: EncodeJson CodePoint where
   encodeJson = encodeJson <<< singleton
 
+instance encodeJsonChar :: EncodeJson Char where
+  encodeJson = encodeJson <<< codePointFromChar
```
Comment (michaelficarra) [inline src/Data/Argonaut/Encode/Class.purs]: You can just use the CodeUnit singleton here.

File: .travis.yml (hunk 2 above)
Comment (CarstenKoenig) [reply .travis.yml]: yes - I'm not sure if I should include this at all - but it should fix the travis build/test for now (once the strongcheck issue got fixed upstream)

File: src/Data/Argonaut/Encode/Class.purs (hunk 3 above)
Comment (CarstenKoenig) [reply src/Data/Argonaut/Encode/Class.purs]: ok thanks - gonna change this

File: src/Data/Argonaut/Decode/Class.purs (hunk 1 above)
Comment (CarstenKoenig) [reply src/Data/Argonaut/Decode/Class.purs]: yes that's possible - I recheck and change (my tooling got broken, so I don't see the warnings right now)

Comment (michaelficarra) [reply src/Data/Argonaut/Decode/Class.purs]: Looks to me like `Data.String` exports everything from `Data.String.CodePoints`: https://github.com/purescript/purescript-strings/blob/80dcf7ef362477214bee9082de091331576feb76/src/Data/String.purs

Comment (CarstenKoenig) [reply src/Data/Argonaut/Decode/Class.purs]: I just changed this one minute ago :D

**Comment by kRITZCREEK** [comment]
Thanks! I cherry picked your changes in #43

**Comment by CarstenKoenig** [comment]
Thank you - glad if I could help out a bit

**Pull request closed without merging**